### DISCUSSION TITLE
Landing Page

### DISCUSSION THREAD

**Discussion by Preet-Sojitra** [ayoayco/astro-reactive-library#112] [Ideas]
Since we have basic landing page. So let's add some content to it and take it to next level. So @ayoayco what are your thoughts and plans?

**Comment by ayo-run** [comment]
Great that you asked, @Preet-Sojitra! Let me give the actual content some thought. I'll get back to you on this thread.

My initial idea was that the header section needs to be updated so that we have two "call to actions" (like in https://mui.com):
1. Get Started (primary button)
2. install text copy field

![Screen Shot 2022-10-19 at 7 15 38 AM](https://user-images.githubusercontent.com/4262489/196603389-c38a7e1a-8055-4053-ac8e-70c988e19974.png)


That could be a good start. Let us also know if you have ideas! Thanks for starting the thread 🧵

**Reply by ayo-run** [reply]
Created an issue for the Call to Action buttons https://github.com/ayoayco/astro-reactive-library/issues/115

**Comment by Lalit3716** [comment]
We should improve responsiveness of the landing page right now it doesn't look good on mobile devices!

**Reply by ayo-run** [reply]
i did not notice this. can you create an issue to describe the task? really good observation.

**Comment by ayo-run** [comment]
By the way we have *some* logos inside `.github/assets` directory. so i guess we can start using those, but they are not good on dark background i think.

So another thing to do could be to make logo variants that are good for light and dark backgrounds.

**Comment by ayo-run** [comment]
I'm writing an issue for the landing page content 🚀

**Comment by Preet-Sojitra** [comment]
Currently `copy to clipboard` symbol is not working. It is just a icon. So why not create a issue regarding that?
![image](https://user-images.githubusercontent.com/87543808/197326309-91cdff23-b269-4b4e-9c7b-0f528bfc7a06.png)

**Reply by ayo-run** [reply]
Ah yeah, good idea. Can you create the issue @Preet-Sojitra 🙏🙏🙏

**Reply by Preet-Sojitra** [reply]
> Ah yeah, good idea. Can you create the issue @Preet-Sojitra 🙏🙏🙏

Sure 👍

**Comment by ayo-run** [comment]
By the way I'm also getting a domain to deploy this when it's ready :)

**Reply by Preet-Sojitra** [reply]
Woaahh!! Nicee 👍

**Reply by ayo-run** [reply]
Launched! https://astro-reactive.dev

**Comment by ayo-run** [comment]
## Landing Page initial content

Hi guys, here's an idea for our landing page content. In issue #132 I want to explore using Solid.js for reactivity, so (being optimistic that it will be successful) I want to communicate this on our landing page. Please let me know what you think!

### Four sections:
1. Main section (the first one you see with the call to action)
2. Astro section
3. Solid.js section
4. Footer

### Main section
1. Copy: Native Astro components for building REACTIVE user interfaces
2. Call to action buttons: "Get Started" and "Copy installation text"
3. Header bar: Logo, "Astro Reactive" title
4. GH stars + social icon links
    - this will turn into a nav bar when the user scrolls down
5. Animated logos of Astro Reactive Library, Astro, and Solid.js
    - animation is: Astro and Solid.js logos revolving around our library's logo

### Astro section
1. Title: "Native Astro Components"
2. one-sentence description, use lorem ipsum for now
3. placeholder for Images/icons indicating benefits of Astro components

### Solid.js section
1. Title: "Reactivity powered by Solid.js"
2. one-sentence description, use lorem ipsum for now
3. placeholder for an image/animation that shows components reacting to events

### Footer
1. show call-to-action buttons
2. GH stars + social icon links

**Reply by ayo-run** [reply]
![image](https://user-images.githubusercontent.com/4262489/197383043-8d4af5f1-918a-4f0c-9523-5b2a1d4da035.jpeg)

**Reply by Lalit3716** [reply]
It looks good one doubt though currently our main section bg is dark but according to your image you want it to be light?

Also I can make Astro section or Solid section.

**Reply by ayo-run** [reply]
Yes I want to start with a light bg Main section.

So the current dark bg will be for the Astro section. I think that is more in line with their branding.

And having a light bg main section communicates a bit more "seriousness" and "reliability" for our library which I want to achieve for targeting enterprises as users.

**Reply by ayo-run** [reply]
If good and clear we can have a separate issue for each section! 🚀

**Reply by Preet-Sojitra** [reply]
> Yes I want to start with a light bg Main section.
> 
> So the current dark bg will be for the Astro section. I think that is more in line with their branding.
> 
> And having a light bg main section communicates a bit more "seriousness" and "reliability" for our library which I want to achieve for targeting enterprises as users.

For main section, have you thought of any color palette for light bg? If not I can work on finding some good color palette for light bg. 
[Huemint](https://huemint.com/) is the tool which I used for finding some good color palette.

**Reply by ayo-run** [reply]
Awesome! @Preet-Sojitra works on the main section. And I think @Lalit3716 said he can take the Astro section. Can you guys create separate issues? Don't worry about  filling up the template you can just link to this discussion.

**Comment by ayo-run** [comment]
I made a mess of our landing page to make it a bit more presentable for now. 🤪

**Reply by ayo-run** [reply]
![image](https://user-images.githubusercontent.com/4262489/200802621-e7d81462-0ba2-4ae4-b32b-1c0b3606197e.jpeg)

https://astro-reactive.dev

**Reply by ayo-run** [reply]
Also doubling back on adding solid as dependency. I'm going for another approach so I closed #132

**Reply by Lalit3716** [reply]
It looks awesome!

**Reply by Lalit3716** [reply]
Also, I think we should give visual feedback to the user when he clicks copy icon, what do you think?

**Reply by ayo-run** [reply]
Yeah I was think the same actually.

**Reply by ayo-run** [reply]
I'm struggling with Tailwind 😅 It took me maybe total of half a day just to get this. Haha

**Reply by Lalit3716** [reply]
I'd say it was worth it, the documentation looks appealing!

**Reply by ayo-run** [reply]
Thank you! Its also gaining some interest over at Twitter: https://twitter.com/ayoayco/status/1589863395465232386?s=46&t=FmnNvdonQZYrHoKi_dVeug

![image](https://user-images.githubusercontent.com/4262489/200821586-990363e2-0fb7-4ef5-bcde-859c178cb76c.jpeg)

**Reply by Lalit3716** [reply]
Awesome!

**Reply by ayo-run** [reply]
Ahhhh I think I broke the copy button thing 😱

**Reply by Lalit3716** [reply]
Ah oh, yup it's not working 😨

**Reply by ayo-run** [reply]
Fixed it! 💪 #187

**Reply by Lalit3716** [reply]
Awesome! now only thing remaining is this #197 for copy button.
